docs/ops/config.md

```diff
@@ -70,6 +70,12 @@ These parameters configure the default HDFS used by Flink. Setups that do not sp
 
 {% include generated/task_manager_configuration.html %}
 
+By default, Flink allocates a fraction of 0.7 of the free memory (total memory configured via taskmanager.heap.mb minus memory used for network buffers) for its managed memory. Managed memory helps Flink to run the batch operators efficiently. It prevents OutOfMemoryExceptions because Flink knows how much memory it can use to execute operations. If Flink runs out of managed memory, it utilizes disk space. Using managed memory, some operations can be performed directly on the raw data without having to deserialize the data to convert it into Java objects. All in all, managed memory improves the robustness and speed of the system.
```

Comment:
Can we get this introductory section auto-generated as well? I see a point when the mentioned defaults change and this section does not.
If not, that's fine as is.

Reply:
Maybe, after mentioning that this helps with batch operators, mention that it is (currently) not used for streaming jobs?

Reply:
We can't auto-generate this section at the moment. There's no place to put this text; you can't have it in the `ConfigGroup` annotation since it wouldn't be a constant (as it relies on other fields for keys/defaults).
We could hard-code the text and place it in `TaskManagerOptions`; it would certainly be more visible than placing it in `config.md`.

Reply:
How about phrasing it like this: `For *batch*  jobs (or when "taskmanager.memory.preallocate" is enabled), Flink allocates by default a fraction ...`